bokeh/tile_providers.py

```diff
@@ -117,10 +169,45 @@ class _TileProvidersModule(types.ModuleType):
         'under %s.'
     )
 
+    class Provider(Enum):
```

Comment:
I'm lukewarm about the name `Provider`, both because `tile_providers.Provider` seems redundant/clunky, but also because I think it might be nice to convey that these are the standard/built-in ones that Bokeh makes available. I haven't thought much about alternatives yet but interested if you have any ideas.

Reply:
I also wasn't too hot on it. Possibly an alternative: `Supplier` or `Maps` or  `Vendor` or `Source` or `TileSource`. (or other combinations like `TileSupplier`). What do you think?

Reply:
of those, I think I like `Vendor` or `Source` the most, but I will think on it some more. Another possible concept to possibly try and get across is "preconfigured"

Reply:
I went for `ThirdParty`not, how do you think that sounds? I think it gets across  that it is preconfigured and a default tile source

Reply:
`ThirdParty` is a bit over-wordy for my taste, I'd try to keep it to a single word rather than two. I think so far `Vendor` or `Vendors` is OK with me